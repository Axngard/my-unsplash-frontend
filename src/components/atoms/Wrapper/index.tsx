import React, { PropsWithChildren } from 'react'
import { screens } from '@src/styles/theme'

/* Styles */
import { WrapperStyled } from './styles'

/* Types */
type Props = PropsWithChildren<{ breakpoint?: string }>

const Wrapper = ({ children, breakpoint = screens.lg }: Props): JSX.Element => {
   return <WrapperStyled theme={{ breakpoint }}>{children}</WrapperStyled>
}

export default Wrapper
