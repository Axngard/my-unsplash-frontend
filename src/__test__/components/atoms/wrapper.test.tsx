import React from 'react'
import { Wrapper } from '@components/atoms'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'

describe('<Wrapper />', () => {
   const children = <h1>Testing</h1>
   const wrapper = shallow(<Wrapper>{children}</Wrapper>)

   it('should render without crashing', () => {
      expect(wrapper).toHaveLength(1)
   })

   it('should match snapshot', () => {
      const wrapper = create(<Wrapper>{children}</Wrapper>)

      expect(wrapper.toJSON()).toMatchSnapshot()
   })

   it('should render children', () => {
      expect(wrapper.find('h1').text()).toEqual('Testing')
   })
})
