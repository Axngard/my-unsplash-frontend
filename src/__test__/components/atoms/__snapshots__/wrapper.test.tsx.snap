// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> should match snapshot 1`] = `
<div
  className="css-zoib7n"
>
  <h1>
    Testing
  </h1>
</div>
`;
